src/lib/y2remote/remote.rb

```diff
@@ -0,0 +1,233 @@
+
+      Yast::Progress.New(caption, " ", steps.size, steps, [], "")
+      Yast::Progress.NextStage
+      Yast::Progress.Title(_("Configuring display manager..."))
+      return false unless configure_display_manager
+
+      if Yast::Mode.normal
+        Yast::Progress.NextStage
+        Yast::Progress.Title(_("Restarting the service..."))
+        restart_services
+        Yast::Progress.NextStage
+      end
+
+      true
+    end
+
+    # Reset the proposal configuration enabling vnc if it was enabled by
+    # linuxrc
+    def reset!
+      @proposed = false
+
+      propose!
+    end
+
+    # It propose the vnc configuration if it has not been proposed yet
+    #
+    # @return [Boolean]
+    def propose!
+      return false if proposed?
+
+      Yast::Linuxrc.vnc ? enable! : disable!
+
+      log.info("Remote Administration was proposed as: #{modes.inspect}")
+
+      @proposed = true
+    end
+
+    # Updates the VNC and xdm configuration
+    #
+    # Called from #write. Ensures that required packages are installed,
+    # enables vnc services and xdm and writes the configuration files,
+    # reporting any error in the process.
+    #
+    # @return [Boolean] true if success, false otherwise
+    def configure_display_manager
+      if enabled?
+        # Install required packages
+        if !Yast::Package.InstallAll(required_packages)
+          log.error "Installing of required packages failed"
+          return false
+        end
+
+        Y2Remote::Modes.all.each { |m| modes.include?(m) ? m.enable! : m.disable! }
+      end
+
+      # Set DISPLAYMANAGER_REMOTE_ACCESS in sysconfig/displaymanager
+      Yast::SCR.Write(
+        Yast.path(".sysconfig.displaymanager.DISPLAYMANAGER_REMOTE_ACCESS"),
+        enabled? ? "yes" : "no"
+      )
+      Yast::SCR.Write(
+        Yast.path(".sysconfig.displaymanager.DISPLAYMANAGER_ROOT_LOGIN_REMOTE"),
+        enabled? ? "yes" : "no"
+      )
+      Yast::SCR.Write(Yast.path(".sysconfig.displaymanager"), nil)
+
+      true
+    end
+
+    def restart_display_manager
+      if Yast::Service.active?(XDM_SERVICE_NAME)
+        Yast::Report.Error(
+          Yast::Message.CannotRestartService(XDM_SERVICE_NAME)
+        ) if !Yast::Service.Reload(XDM_SERVICE_NAME)
+
+        Yast::Report.Warning(
+          _(
+            "Your display manager must be restarted.\n" \
+            "To take the changes in remote administration into account, \n" \
+            "please restart it manually or log out and log in again."
+          )
+        )
+      elsif !Yast::Service.Restart(XDM_SERVICE_NAME)
+        Yast::Report.Error(
+          Yast::Message.CannotRestartService(XDM_SERVICE_NAME)
+        )
+      end
+    end
+
+    # Restarts services, reporting errors to the user
+    def restart_services
+      Yast::SystemdTarget.set_default(GRAPHICAL_TARGET) if enabled?
+
+      Y2Remote::Modes.all { |m| modes.include?(m.to_sym) ? m.restart! : m.stop! }
```

Comment:
You could implement this logic in a class method. Something like: `restart_modes(modes)` (please, find a better name :) ).

Reply:
true :)